src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,199 @@
+                    relinkNewNode(tempNode, value);
+                }
+                tempNode = tempNode.next;
+
+            }
+        } else {
+            int indexCounter = size;
+            Node<T> tempNode = tale;
+            while (indexCounter != index) {
+                indexCounter--;
+                if (indexCounter == index) {
+                    relinkNewNode(tempNode, value);
+                }
+                tempNode = tempNode.prev;
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexForBound(index, size - 1);
+        return getNode(index).body;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexForBound(index, size - 1);
+        if (index == 0) {
+            Node<T> initialHeadNode = head;
+            head = new Node<>(null, value, initialHeadNode.next);
+            initialHeadNode.next.prev = head;
+            return initialHeadNode.body;
+        } else if (index == size - 1) {
+            Node<T> initialTaleNode = tale;
+            tale = new Node<>(initialTaleNode.prev, value, null);
+            initialTaleNode.prev.next = tale;
+            return initialTaleNode.body;
+        } else {
+            Node<T> toRemoveNode = getNode(index);
+            Node<T> insteadNode = new Node<>(toRemoveNode.prev, value, toRemoveNode.next);
+            toRemoveNode.prev.next = insteadNode;
+            toRemoveNode.next.prev = insteadNode;
+            return toRemoveNode.body;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexForBound(index, size - 1);
+        Node<T> toRemoveNode = getNode(index);
+        if (index == 0) {
+            head = getNode(1);
+        } else if (index == size - 1) {
+            tale = getNode(size - 2);
+        }
+        unlinkNode(toRemoveNode);
+        size--;
+        return toRemoveNode.body;
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < size; i++) {
+            if (get(i) == object || get(i) != null && get(i).equals(object)) {
+                remove(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (size != 0) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private void relinkNewNode(Node<T> tempNode, T value) {
+        Node<T> toAdd = new Node<>(tempNode.prev, value, tempNode);
+        tempNode.prev.next = toAdd;
+        tempNode.prev = toAdd;
+        size++;
+        return;
+    }
+
+    private void checkIndexForBound(int index, int specialSize) {
```

Comment:
it's better to have 2 check methods. one for add and second for other cases. in such way you won't need to pass specialSize in parameters